body-tracking-samples/simple_3d_viewer/main.cpp

```diff
@@ -99,24 +103,172 @@ bool ParseInputSettingsFromArg(int argc, char** argv, InputSettings& inputSettin
         {
             inputSettings.CpuOnlyMode = true;
         }
+        else if (inputArg == std::string("OFFLINE"))
+        {
+            inputSettings.Offline = true;
+        }
+     
         else
         {
-            return false;
+            inputSettings.FileName = inputArg;
```

Comment:
Thank you for submitting this pull request!

I would say the contract for the OFFLINE setting would be to ensure that there is always a file name after it. So the if statement at 106 should validate argc also has a file name and it should immediate read it rather than go back up to the top of the for loop. And if the command is malformed we should exit with an error rather than run the wrong command.

Reply:
Thank you - I have made this change. The only caveat is that it will stop reading arguments after it finds OFFLINE and a file name, so these should be last, so if the user also wants to use CPU mode, this flag should be specified first.